src/common/stats/StatsManager.cpp

```diff
@@ -311,6 +323,57 @@ StatusOr<StatsManager::VT> StatsManager::readHisto(const std::string& counterNam
     return sm.histograms_[index].second->getPercentileEstimate(pct, level);
 }
 
+
+// static
+StatusOr<StatsManager::ParsedName>
+StatsManager::parseMetricName(folly::StringPiece metricName) {
+    std::vector<std::string> parts;
+    folly::split(".", metricName, parts, true);
+    if (parts.size() != 3) {
+        LOG(ERROR) << "\"" << metricName << "\" is not a valid metric name";
+        return Status::Error("\"%s\" is not a valid metric name", metricName.data());
+    }
+
+    std::string name = parts[0];
+    TimeRange range;
+    if (parts[2] == "60") {
+        range = TimeRange::ONE_MINUTE;
+    } else if (parts[2] == "600") {
+        range = TimeRange::TEN_MINUTES;
+    } else if (parts[2] == "3600") {
+        range = TimeRange::ONE_HOUR;
+    } else {
+        // Unsupported time range
+        LOG(ERROR) << "Unsupported time range \"" << parts[2] << "\"";
+        return Status::Error(folly::stringPrintf("Unsupported time range \"%s\"",
+                                                 parts[2].c_str()));
+    }
+
+    StatsMethod method;
+    folly::toLowerAscii(parts[1]);
+    if (parts[1] == "sum") {
+        method = StatsMethod::SUM;
+    } else if (parts[1] == "count") {
+        method = StatsMethod::COUNT;
+    } else if (parts[1] == "avg") {
+        method = StatsMethod::AVG;
+    } else if (parts[1] == "rate") {
+        method = StatsMethod::RATE;
+    // } else if (parts[1][0] == 'p') {   // TODO(shylock)
```

Comment:
why not support it？

Reply:
The method is not used by promehteus, this just keep compatible with previous code.